File: Sanskrit/app/src/main/java/com/example/sanskrit/MainActivity.java
package com.example.sanskrit;


import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.tabs.TabItem;
import com.google.android.material.tabs.TabLayout;

public class MainActivity extends AppCompatActivity {

    private ViewPager viewPager;
    private TabLayout tabLayout;
    private TabItem numbers , family, colors, phrases;
private PageAdapter pageAdapter;


    @Override
    protected void onCreate(Bundle savedInstanceState) {


        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        tabLayout = (TabLayout) findViewById(R.id.tablayout);
        viewPager = (ViewPager) findViewById(R.id.vpager);


        pageAdapter = new PageAdapter(this,getSupportFragmentManager());

        viewPager.setAdapter(pageAdapter);

        tabLayout.addTab(tabLayout.newTab().setText("Number"));
        tabLayout.addTab(tabLayout.newTab().setText("Family"));
        tabLayout.addTab(tabLayout.newTab().setText("Colors"));
        tabLayout.addTab(tabLayout.newTab().setText("Phrases"));
        tabLayout.setTabGravity(TabLayout.GRAVITY_FILL);
        tabLayout.setupWithViewPager(viewPager);


    }
}

File: QuotesApp/app/src/main/java/com/example/quotesapp/MainActivity.kt
package com.example.quotesapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val author : TextView =findViewById(R.id.author)
        val quote : TextView = findViewById(R.id.quote)

        val quotesAPI = RetrofitHelper.getInstance().create(QuotesApi::class.java)
        GlobalScope.launch {
            val result = quotesAPI.getQuotes()

            val quotesList = result.body()
            runOnUiThread(Runnable {
                if (quotesList != null) {
                    author.text = quotesList.author
                    quote.text = quotesList.en
                }
                 })
        }

    }
}